src/app/clusters/color-control-server/color-control-server.cpp

```diff
@@ -371,11 +395,26 @@ class DefaultColorControlSceneHandler : public scenes::DefaultSceneHandlerImpl
         }
     }
 
-    void AddAttributeValuePair(ScenesManagement::Structs::AttributeValuePair::Type * pairs, AttributeId id, uint32_t value,
+    /// AddAttributeValuePair
+    /// @brief Helper function to add an attribute value pair to the attribute value pair array in the color control SceneHandler
+    /// @param pairs list of attribute value pairs
+    /// @param id attribute id
+    /// @param value attribute value
+    /// @param attributeCount number of attributes in the list, incremented by this function, used to keep track of how many
+    /// attributes from the array are being used for the list to encode
+    template <typename Type>
+    void AddAttributeValuePair(ScenesManagement::Structs::AttributeValuePair::Type * pairs, AttributeId id, Type value,
                                size_t & attributeCount)
     {
-        pairs[attributeCount].attributeID    = id;
-        pairs[attributeCount].attributeValue = value;
+        pairs[attributeCount].attributeID = id;
+        if constexpr (sizeof(Type) == sizeof(uint8_t))
```

Comment:
however we also have signed/unsigned diffs, so actual selection seems to need more than just sizeof.

Would we need explicit template specialization instead? Or some other approach?

Reply:
This helper is made specifically for the the level control scene handler, which only has unsigned scenable attributes. Since handler implementation differs a lot between clusters, I do not think we need to generalise more for this.

Reply:
We are putting the cluster back to provisional, so backwards compatibility is not required anymore.